jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceCollection.java

```diff
@@ -126,8 +126,8 @@ public Resource resolve(String subUriPath)
         for (Resource res : _resources)
         {
             addedResource = res.resolve(subUriPath);
-            if (!addedResource.exists())
-                continue;
+            if (addedResource == null)
```

Comment:
I think more changes below  are needed for code clarity.

specifically if resources == null, can we just explicitly return null?